src/impl/peerconnection.cpp

```diff
@@ -73,11 +73,6 @@ PeerConnection::PeerConnection(Configuration config_)
 	}
 }
 
-PeerConnection::~PeerConnection() {
-	PLOG_VERBOSE << "Destroying PeerConnection";
-	mProcessor.join();
```

Comment:
I'm afraid joining here may also create a deadlock situation if the user destroys the `PeerConnection` from a callback, either by releasing the last shared_ptr or by calling `rtcDeletePeerConnection` if using the C API.

I think what your are trying to do might not be feasible: you can't wait for the following tasks to finish from one of the tasks as doing so you block the task processor and therefore the following tasks.